lib/dateTime.js

```diff
@@ -0,0 +1,578 @@
+	};
+
+	let escape = false;
+	let buffer = '';
+	let value = '';
+	for (let i = 0; i < pattern.length; i++) {
+		const c = pattern.charAt(i);
+		if (c === "'") {
+			if (!escape) {
+				value += doReplacements(buffer);
+				buffer = '';
+			}
+			escape = !escape;
+		} else if (escape) {
+			value += c;
+		} else {
+			buffer += c;
+		}
+	}
+	value += doReplacements(buffer);
+
+	return value;
+
+}
+
+export function getDateTimeDescriptor() {
+
+	const language = getLanguage();
+
+	const subtags = language.split('-');
+	const baseLanguage = subtags[0];
+
+	let hour24 = (hour24locales.indexOf(baseLanguage) > -1);
+	if (language === 'zh-tw') {
+		hour24 = false;
+	}
+
+	const timeFormat = getTimeFormat(hour24, language, baseLanguage);
+
+	let dateFormats = ['dddd, MMMM d, yyyy', 'MMM d, yyyy', 'M/d/yyyy', 'MMMM yyyy', 'MMMM d'];
+	let fullTimeFormat = `${timeFormat} ZZZ`;
+	let dayPeriods = ['AM', 'PM'];
+	let months = [
+		['January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'],
+		['Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sep', 'Oct', 'Nov', 'Dec']
+	];
+	let days = [
+		['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'],
+		['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'],
+		['S', 'M', 'T', 'W', 'T', 'F', 'S']
+	];
+	let firstDayOfWeek = (mondayFirstDayLocales.indexOf(baseLanguage) > -1) ? 1 : 0;
+	let weekendStartDay = 6;
+	let weekendEndDay = 0;
+
+	switch (baseLanguage) {
+		case 'ar':
+			dateFormats = ['dddd, d MMMM, yyyy', 'dd MMMM, yyyy', 'dd/MM/yyyy', 'MMMM, yyyy', 'd MMMM'];
+			dayPeriods = ['ص', 'م'];
+			months[0] = months[1] = ['يناير', 'فبراير', 'مارس', 'أبريل', 'مايو', 'يونيو', 'يوليو', 'أغسطس', 'سبتمبر', 'أكتوبر', 'نوفمبر', 'ديسمبر'];
+			days = [
+				['الأحد', 'الإثنين', 'الثلاثاء', 'الأربعاء', 'الخميس', 'الجمعة', 'السبت'],
+				['أحد', 'إثنين', 'ثلاثاء', 'أربعاء', 'خميس', 'جمعة', 'سبت'],
+				['أ', 'إ', 'ث', 'أر', 'خ', 'ج', 'س']
+			];
+			firstDayOfWeek = 6;
+			weekendStartDay = 4;
+			weekendEndDay = 5;
+			break;
+		case 'da':
+			dateFormats = ['dddd \'den\' d. MMMM yyyy', 'd. MMM. yyyy', 'dd/MM/yyyy', 'MMMM yyyy', 'd. MMMM'];
+			months = [numericMonths, numericMonths];
+			days = [
+				['søndag', 'mandag', 'tirsdag', 'onsdag', 'torsdag', 'fredag', 'lørdag'],
+				['søn.', 'man.', 'tir.', 'ons.', 'tor.', 'fre.', 'lør.'],
+				['S', 'M', 'T', 'O', 'T', 'F', 'L']
+			];
+			break;
+		case 'de':
+			dateFormats = ['dddd d. MMMM yyyy', 'd. MMMM yyyy', 'dd-MM-yyyy', 'MMMM yyyy', 'd. MMMM'];
+			months = [numericMonths, numericMonths];
+			days = [
+				['Sonntag', 'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag'],
+				['So.', 'Mo.', 'Di.', 'Mi.', 'Do.', 'Fr.', 'Sa.'],
+				['S', 'M', 'D', 'M', 'D', 'F', 'S']
+			];
+			break;
+		case 'en':
+			break;
+		case 'es':
+			dateFormats = ['dddd d\' de \'MMMM\' de \'yyyy', 'd\' de \'MMMM\' de \'yyyy', 'dd/MM/yyyy', 'MMMM yyyy', 'd\' de \'MMMM'];
```

Comment:
Looks like this is in the same boat with `es` to `es-MX` differences (`'d/M/yyyy'` vs `'dd/MM/yyyy'`).  Worth adding the proper `es` value?

Reply:
Fixed.